### ISSUE TITLE
Slicing torch.Size returns tuple

### ISSUE THREAD

**Issue by szagoruyko** [pytorch/pytorch#730]
E.g. this: `torch.Tensor(torch.Size((256,3,9,9,2))[:-1])` will return:
```
 256
   3
   9
   9
[torch.FloatTensor of size 4]
```
Slicing `torch.Size` should return `torch.Size`.

**Comment by artcg** [comment]
I think you just need to add to torch.Size:

```python
def __getitem__(self, *args):
    return Size(tuple.__getitem__(self, *args))
```

**Comment by apaszke** [comment]
It's not that simple because torch.Size is defined in C. I've already fixed that and will push it later.

**Comment by apaszke** [comment]
Fixed in #752.
